add(epp): ceph rest api (#1523)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/applications-ceph-restapi.md

---
id: applications-ceph-restapi
title: Ceph Rest API
---
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

## Pack Assets

### Templates

The Centreon Pack Ceph brings a host template:
* App-Ceph-Restapi-custom

> It brings the following Service Templates:

Reviewer: ```suggestion
It brings the following service templates:
```

| Health         | App-Ceph-Restapi-Health | X       |           |
| Osd            | App-Ceph-Restapi-Osd    | X       |           |
| Pools          | App-Ceph-Restapi-Pools  |         | X         |

### Discovery rules

| Rule name                  | Description                            |
|:---------------------------|:---------------------------------------|
| App-Ceph-Restapi-Pool-Name | Discover pools and monitor utilization |

### Collected metrics & status

<Tabs groupId="sync">
<TabItem value="Health" label="Health">

| Metric name   | Description                    | Unit  |
| :------------ | :----------------------------- | :---- |
| health status | Current overall cluster status |       |

</TabItem>
<TabItem value="Osd" label="Osd">

| Metric name                  | Description                                                 | Unit  |
| :--------------------------- | :---------------------------------------------------------- | :---- |
| osd.detected.count           | Number of online storage daemons detected                   |       |
| osd.online.count             | Number of online storage daemons in Ceph cluster            |       |

> | osd.online.percentage        | Pourcentage of online storage daemons in Ceph cluster       | %     |

Reviewer: ```suggestion
| osd.online.percentage        | Perrcentage of online storage daemons in Ceph cluster       | %     |
```

| osd.participating.count      | Number of participating storage daemons in Ceph cluster     |       |
| osd.participating.percentage | Percentage of participating storage daemons in Ceph cluster | %     |

</TabItem>
<TabItem value="Pools" label="Pools">

| Metric name                                 | Description                          | Unit  |
| :------------------------------------------ | :----------------------------------- | :---- |
| pools.detected.count                        | Number of pools detected             |       |

> | *pool_name*#pool.space.usage.bytes          | Space used on the pool |             |       |

Reviewer: ```suggestion
| *pool_name*#pool.space.usage.bytes          | Space used on the pool |                     |
```

| *pool_name*#pool.space.free.bytes           | Free space left on the pool          |       |
| *pool_name*#pool.space.usage.percentage     | Space used on the pool in percentage | %     |
| *pool_name*#pool.read.usage.bytespersecond  | Read rate                            | B/s   |
| *pool_name*#pool.write.usage.bytespersecond | Write rate                           | B/s   |

</TabItem>
</Tabs>

## Prerequisites

> To monitor your Ceph, an user with read privileges is needed:

Reviewer: ```suggestion
To monitor your Ceph, a user with read privileges is needed:
```

> To monitor your Ceph, a user with read privileges is needed:
> E.g: hhttps://docs.ceph.com/en/latest/mgr/ceph_api/

Reviewer 2: ```suggestion
To monitor your Ceph, a user with read privileges is required. Please refer to their official documentation: https://docs.ceph.com/en/latest/mgr/ceph_api/
```
